cms/src/main/java/uk/nhs/digital/s3/S3ConnectorImpl.java

```diff
@@ -0,0 +1,85 @@
+package uk.nhs.digital.s3;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.SystemPropertiesCredentialsProvider;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.AccessControlList;
+import com.amazonaws.services.s3.model.GroupGrantee;
+import com.amazonaws.services.s3.model.Permission;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+
+/**
+ * Implement S3Connector to allow uploading files to S3 and managing ACL.
+ */
+public class S3ConnectorImpl implements S3Connector {
+
+    private String bucketName;
+    private Region region;
+    private AmazonS3 s3;
+
+    public S3ConnectorImpl(String bucketName, String region) {
+        this.bucketName = bucketName;
+        this.region = Region.getRegion(Regions.fromName(region));
+
+        connect();
+    }
+
+    public boolean publishResource(String objectPath) {
+        AccessControlList acl = s3.getObjectAcl(bucketName, objectPath);
+        acl.grantPermission(GroupGrantee.AllUsers, Permission.Read);
+        s3.setObjectAcl(bucketName, objectPath, acl);
+
+        return true;
+    }
+
+    public boolean unpublishResource(String objectPath) {
+        AccessControlList acl = s3.getObjectAcl(bucketName, objectPath);
+        acl.revokeAllPermissions(GroupGrantee.AllUsers);
+        s3.setObjectAcl(bucketName, objectPath, acl);
+
+        return true;
+    }
+
+    public boolean uploadFile(String objectPath) {
+        s3.putObject(new PutObjectRequest(bucketName, objectPath, createSampleFile()));
+
+        return true;
+    }
+
+    protected void connect() {
+        AWSCredentialsProvider provider = new SystemPropertiesCredentialsProvider();
+        s3 = new AmazonS3Client(provider);
```

Comment:
It seems that it would be better to move this entire initialisation, together with the 'region' field (which is only needed for initialisation) outside of this class and into `uk.nhs.digital.modules.S3ConnectorServiceRegistrationModule#doInitialize`. An instance of `AmazonS3Client` (the `s3` reference could then be provided to this class through its constructor.